src/_data/data.js

```diff
@@ -598,11 +638,49 @@ function processAllContent() {
   const rootList = lists.find(list => list.id === ROOT_LIST_ID);
 
   crossReferenceFaqsAndGuidanceRequests(faqs, guidanceRequests);
+
+  // Create dynamic lists BEFORE cross-referencing so they can be referenced in YAML files
+  const dynamicLists = DYNAMIC_LISTS.map(createDynamicList);
+
+  // Add dynamic lists to the lists array so they can be found during cross-referencing
+  lists.push(...dynamicLists);
+
+  // Add dynamic list IDs to root's YAML (only those not manually placed)
+  // Use ~dynamic/ namespace to distinguish from regular lists
+  // Separate into top and bottom insertions based on insertAt property
+  const topInsertions = [];
+  const bottomInsertions = [];
+
+  DYNAMIC_LISTS.forEach(config => {
+    const reference = `~dynamic/${config.id}`;
+    // Only auto-add if not already manually placed
+    if (!rootList.yaml.faqs.includes(reference)) {
+      if (config.insertAt === 'top') {
+        topInsertions.push(reference);
+      } else {
+        bottomInsertions.push(reference);
+      }
+    }
+  });
+
+  // Add top insertions in reverse order (so first in array appears first in list)
+  topInsertions.reverse().forEach(reference => {
+    rootList.yaml.faqs.unshift(reference);
+  });
+
```

Comment:
I believe this should work:
```suggestion
  rootList.yaml.faqs.unshift(...topInsertions.reverse());

```